---
title: "RaggedExperiment: Object Size Comparisons"
author: "Waldron Lab"
date: "`r format(Sys.time(), '%B %d, %Y')`"
vignette: >
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteIndexEntry{RaggedExperiment Object Size Comparisons}
  %\VignetteEncoding{UTF-8}
output:
  BiocStyle::html_document:
    number_sections: yes
    toc: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, cache = TRUE)
```
# GitHub Repository

https://github.com/LiNk-NY/RaggedExperiment_SoftNote cloned and updated from
https://github.com/vjcitn/raggedchk

# Installation

```{r,eval=FALSE}
if (!require("BiocManager"))
    install.packages("BiocManager")
BiocManager::install("RaggedExperiment")
```

```{r,include=TRUE,results="hide",message=FALSE,warning=FALSE}
library(RaggedExperiment)
library(curatedTCGAData)
library(GenomeInfoDb)
library(RaggedExperiment.SoftNote)
```

```{r,message=FALSE,cache=TRUE}
BRCA <- curatedTCGAData(
    "BRCA", version = "2.0.1", assays = c("CNASeq", "Mutation"), dry.run = FALSE
)
BRCA
```

## Measuring size

### CNAseq

```{r,eval=FALSE}
object_size(BRCA[["BRCA_CNASeq-20160128"]])
object_size(sparseAssay(BRCA[["BRCA_CNASeq-20160128"]], sparse = TRUE))
object_size(compactAssay(BRCA[["BRCA_CNASeq-20160128"]]))
object_size(sparseAssay(BRCA[["BRCA_CNASeq-20160128"]]))
```

### Mutation

```{r,eval=FALSE}
object_size(BRCA[["BRCA_Mutation-20160128"]])
# object_size(sparseAssay(BRCA[["BRCA_Mutation-20160128"]], sparse = TRUE))
object_size(compactAssay(BRCA[["BRCA_Mutation-20160128"]]))
object_size(sparseAssay(BRCA[["BRCA_Mutation-20160128"]]))
```

# Object Sizes from `curatedTCGAData`

```{r,include=TRUE,results="hide",message=FALSE,warning=FALSE}
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
```

## Extract all gene regions from TxDb

```{r}
txdb <- TxDb.Hsapiens.UCSC.hg19.knownGene
genes <- genes(txdb, single.strand.genes.only = FALSE)
genes <- keepStandardChromosomes(genes, pruning.mode = "coarse")
ugenes <- unlist(genes)
```

## Standardize `seqlevelsStyle` to UCSC

```{r}
# https://gdac.broadinstitute.org/runs/analyses__2016_01_28/reports/cancer/BRCA-TP/CopyNumberLowPass_Gistic2/nozzle.html
re <- BRCA[["BRCA_CNASeq-20160128"]]
## possible bug -- seqlevelsStyle must go first, then genome
seqlevelsStyle(re) <- "UCSC"
genome(rowRanges(re)) <- "hg19"
rowRanges(re)
```

## It looks like the order matters, otherwise you get NCBI seqlevels...

```{r,eval=FALSE}
re2 <- BRCA[["BRCA_CNASeq-20160128"]]
genome(rowRanges(re2)) <- "hg19"
seqlevelsStyle(re2) <- "UCSC"
rowRanges(re2)
identical(re, re2)
#' [1] FALSE
```

## CNAseq in genes

```{r}
ingenes <- subsetByOverlaps(re, ugenes)
```

```{r,eval=FALSE}
object_size(compactAssay(ingenes))
object_size(sparseAssay(ingenes))
```

## Mutations in genes

```{r}
# https://gdac.broadinstitute.org/runs/analyses__2016_01_28/reports/cancer/BRCA-TP/CopyNumberLowPass_Gistic2/nozzle.html
mre <- BRCA[["BRCA_Mutation-20160128"]]
## possible bug -- seqlevelsStyle must go first, then genome
seqlevelsStyle(mre) <- "UCSC"
genome(rowRanges(mre)) <- "hg19"
rowRanges(mre)
```

```{r}
mingenes <- subsetByOverlaps(mre, ugenes)
```

```{r,eval=FALSE}
object_size(compactAssay(mingenes))
object_size(sparseAssay(mingenes))
```

## Obtaining data directly from `RTCGAToolbox`

```{r}
library(RTCGAToolbox)
getLinks("BRCA", CNASeq = TRUE)
BRCAseq <- getFirehoseData("BRCA", CNASeq = TRUE)
cnatoolbox <- biocExtract(BRCAseq, "CNASeq")
seqlevelsStyle(cnatoolbox) <- "UCSC"
genome(cnatoolbox) <- "hg19"
rowRanges(cnatoolbox)

getLinks("BRCA", Mutation = TRUE)
BRCAmut <- getFirehoseData("BRCA", Mutation = TRUE)
muttoolbox <- biocExtract(BRCAmut, "Mutation")
seqlevelsStyle(muttoolbox) <- "UCSC"
genome(muttoolbox) <- "hg19"
rowRanges(muttoolbox)
```

## Sizes for CNAseq from `RTCGAToolbox`

```{r,eval=FALSE}
object_size(cnatoolbox)
object_size(BRCAseq@CNASeq)
object_size(sparseAssay(cnatoolbox, sparse = TRUE))
object_size(compactAssay(cnatoolbox))
object_size(sparseAssay(cnatoolbox))
```


```{r,eval=FALSE}
object_size(muttoolbox)
object_size(BRCAmut@Mutation)
# object_size(sparseAssay(muttoolbox, sparse = TRUE)) # typeof character
object_size(compactAssay(muttoolbox))
object_size(sparseAssay(muttoolbox))
```

## Restrict to genic regions

```{r}
incnabox <- subsetByOverlaps(cnatoolbox, ugenes)
inmutbox <- subsetByOverlaps(muttoolbox, ugenes)
```


```{r,eval=FALSE}
object_size(incnabox)
# NA
object_size(sparseAssay(incnabox, sparse = TRUE))
object_size(compactAssay(incnabox))
object_size(sparseAssay(incnabox))
```

```{r,eval=FALSE}
object_size(inmutbox)
# NA
# object_size(sparseAssay(inmutbox, sparse = TRUE))
object_size(compactAssay(inmutbox))
object_size(sparseAssay(inmutbox))
```

# Table of object sizes by data type and source

```{r,echo=FALSE}
colnames <- c(
  "Data Source", "Assay", "Data Type", "RaggedExperiment",
  "as.data.frame", "sparse Matrix", "matrix (reduced rows)", "matrix (sparse)"
)
adata <- data.frame(
  rep("curatedTCGAData", 4L),
  c("CNASeq", "CNASeq (in genes)", "Mutation", "Mutation (in genes)"),
  c("numeric", "numeric", "character", "character")
)
tdata <- matrix(c(
  c(
    object_size(BRCA[["BRCA_CNASeq-20160128"]]),
    object_size(as.data.frame(BRCA[["BRCA_CNASeq-20160128"]]@assays)) ,
    object_size(sparseAssay(BRCA[["BRCA_CNASeq-20160128"]], sparse = TRUE)), 
    object_size(compactAssay(BRCA[["BRCA_CNASeq-20160128"]])), 
    object_size(sparseAssay(BRCA[["BRCA_CNASeq-20160128"]])) 
  ),
  c(
    object_size(ingenes),
    object_size(as.data.frame(ingenes@assays)),
    object_size(sparseAssay(ingenes, sparse = TRUE)),
    object_size(compactAssay(ingenes)),
    object_size(sparseAssay(ingenes))
  ),
  c(
    object_size(BRCA[["BRCA_Mutation-20160128"]]),
    object_size(as.data.frame(BRCA[["BRCA_Mutation-20160128"]]@assays)),
    NA_character_,
    object_size(compactAssay(BRCA[["BRCA_Mutation-20160128"]])),
    object_size(sparseAssay(BRCA[["BRCA_Mutation-20160128"]]))
  ),
  c(
    object_size(mingenes),
    object_size(as.data.frame(mingenes@assays)),
    NA_character_,
    object_size(compactAssay(mingenes)),
    object_size(sparseAssay(mingenes))
  )
), nrow = 4, byrow = TRUE)
tdata <- as.data.frame(tdata)
alldf <-cbind.data.frame(adata, tdata)
colnames(alldf) <- colnames
knitr::kable(alldf, "pipe")
```
