<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>RaggedExperiment: A Use Case • RaggedExperiment.SoftNote</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="RaggedExperiment: A Use Case">
<meta property="og:description" content="RaggedExperiment.SoftNote">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RaggedExperiment.SoftNote</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/RaggedExperiment_Manuscript_Example.html">RaggedExperiment: A Use Case</a>
    </li>
    <li>
      <a href="../articles/RaggedExperiment_Manuscript_qReduceAssay.html">qReduceAssay BRCA</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>RaggedExperiment: A Use Case</h1>
                        <h4 data-toc-skip class="author">Waldron
Lab</h4>
            
            <h4 data-toc-skip class="date">March 16, 2023</h4>
      
      
      <div class="hidden name"><code>RaggedExperiment_Manuscript_Example.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="github-repository">GitHub Repository<a class="anchor" aria-label="anchor" href="#github-repository"></a>
</h2>
<p><a href="https://github.com/LiNk-NY/RaggedExperiment_SoftNote" class="external-link uri">https://github.com/LiNk-NY/RaggedExperiment_SoftNote</a>
cloned and updated from <a href="https://github.com/vjcitn/raggedchk" class="external-link uri">https://github.com/vjcitn/raggedchk</a></p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"RaggedExperiment"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">RaggedExperiment</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">curatedTCGAData</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomeInfoDb" class="external-link">GenomeInfoDb</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">RaggedExperiment.SoftNote</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">BRCA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/curatedTCGAData/man/curatedTCGAData.html" class="external-link">curatedTCGAData</a></span><span class="op">(</span></span>
<span>    <span class="st">"BRCA"</span>, version <span class="op">=</span> <span class="st">"2.0.1"</span>, assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CNASeq"</span>, <span class="st">"Mutation"</span><span class="op">)</span>, dry.run <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="va">BRCA</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 2 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 2:</span></span>
<span><span class="co">##  [1] BRCA_CNASeq-20160128: RaggedExperiment with 5298 rows and 38 columns</span></span>
<span><span class="co">##  [2] BRCA_Mutation-20160128: RaggedExperiment with 90490 rows and 993 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<div class="section level3">
<h3 id="measuring-size">Measuring size<a class="anchor" aria-label="anchor" href="#measuring-size"></a>
</h3>
<div class="section level4">
<h4 id="cnaseq">CNAseq<a class="anchor" aria-label="anchor" href="#cnaseq"></a>
</h4>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="va">BRCA</span><span class="op">[[</span><span class="st">"BRCA_CNASeq-20160128"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RaggedExperiment/man/assay-functions.html" class="external-link">sparseAssay</a></span><span class="op">(</span><span class="va">BRCA</span><span class="op">[[</span><span class="st">"BRCA_CNASeq-20160128"</span><span class="op">]</span><span class="op">]</span>, sparse <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RaggedExperiment/man/assay-functions.html" class="external-link">compactAssay</a></span><span class="op">(</span><span class="va">BRCA</span><span class="op">[[</span><span class="st">"BRCA_CNASeq-20160128"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RaggedExperiment/man/assay-functions.html" class="external-link">sparseAssay</a></span><span class="op">(</span><span class="va">BRCA</span><span class="op">[[</span><span class="st">"BRCA_CNASeq-20160128"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="mutation">Mutation<a class="anchor" aria-label="anchor" href="#mutation"></a>
</h4>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="va">BRCA</span><span class="op">[[</span><span class="st">"BRCA_Mutation-20160128"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co"># object_size(sparseAssay(BRCA[["BRCA_Mutation-20160128"]], sparse = TRUE))</span></span>
<span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RaggedExperiment/man/assay-functions.html" class="external-link">compactAssay</a></span><span class="op">(</span><span class="va">BRCA</span><span class="op">[[</span><span class="st">"BRCA_Mutation-20160128"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RaggedExperiment/man/assay-functions.html" class="external-link">sparseAssay</a></span><span class="op">(</span><span class="va">BRCA</span><span class="op">[[</span><span class="st">"BRCA_Mutation-20160128"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="object-sizes-from-curatedtcgadata">Object Sizes from <code>curatedTCGAData</code><a class="anchor" aria-label="anchor" href="#object-sizes-from-curatedtcgadata"></a>
</h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="extract-all-gene-regions-from-txdb">Extract all gene regions from TxDb<a class="anchor" aria-label="anchor" href="#extract-all-gene-regions-from-txdb"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">txdb</span> <span class="op">&lt;-</span> <span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">txdb</span>, single.strand.genes.only <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevels-wrappers.html" class="external-link">keepStandardChromosomes</a></span><span class="op">(</span><span class="va">genes</span>, pruning.mode <span class="op">=</span> <span class="st">"coarse"</span><span class="op">)</span></span>
<span><span class="va">ugenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="standardize-seqlevelsstyle-to-ucsc">Standardize <code>seqlevelsStyle</code> to UCSC<a class="anchor" aria-label="anchor" href="#standardize-seqlevelsstyle-to-ucsc"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># https://gdac.broadinstitute.org/runs/analyses__2016_01_28/reports/cancer/BRCA-TP/CopyNumberLowPass_Gistic2/nozzle.html</span></span>
<span><span class="va">re</span> <span class="op">&lt;-</span> <span class="va">BRCA</span><span class="op">[[</span><span class="st">"BRCA_CNASeq-20160128"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">## possible bug -- seqlevelsStyle must go first, then genome</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevelsStyle.html" class="external-link">seqlevelsStyle</a></span><span class="op">(</span><span class="va">re</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"UCSC"</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">genome</a></span><span class="op">(</span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">re</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"hg19"</span></span>
<span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">re</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 5298 ranges and 0 metadata columns:</span></span>
<span><span class="co">##          seqnames              ranges strand</span></span>
<span><span class="co">##             &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">##      [1]     chr1       10209-2583075      *</span></span>
<span><span class="co">##      [2]     chr1   2583076-249240606      *</span></span>
<span><span class="co">##      [3]     chr2     10002-243189359      *</span></span>
<span><span class="co">##      [4]     chr3     60175-162511435      *</span></span>
<span><span class="co">##      [5]     chr3 162511436-162626067      *</span></span>
<span><span class="co">##      ...      ...                 ...    ...</span></span>
<span><span class="co">##   [5294]    chr20      60001-62965506      *</span></span>
<span><span class="co">##   [5295]    chr21    9422166-48119869      *</span></span>
<span><span class="co">##   [5296]    chr22   16051206-51244552      *</span></span>
<span><span class="co">##   [5297]    chr23   2699503-116067549      *</span></span>
<span><span class="co">##   [5298]    chr24    2649450-28784074      *</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 24 sequences from hg19 genome; no seqlengths</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="it-looks-like-the-order-matters-otherwise-you-get-ncbi-seqlevels">It looks like the order matters, otherwise you get NCBI
seqlevels…<a class="anchor" aria-label="anchor" href="#it-looks-like-the-order-matters-otherwise-you-get-ncbi-seqlevels"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">re2</span> <span class="op">&lt;-</span> <span class="va">BRCA</span><span class="op">[[</span><span class="st">"BRCA_CNASeq-20160128"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">genome</a></span><span class="op">(</span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">re2</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"hg19"</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevelsStyle.html" class="external-link">seqlevelsStyle</a></span><span class="op">(</span><span class="va">re2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"UCSC"</span></span>
<span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">re2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/identical.html" class="external-link">identical</a></span><span class="op">(</span><span class="va">re</span>, <span class="va">re2</span><span class="op">)</span></span>
<span><span class="co">#' [1] FALSE</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="cnaseq-in-genes">CNAseq in genes<a class="anchor" aria-label="anchor" href="#cnaseq-in-genes"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ingenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">re</span>, <span class="va">ugenes</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RaggedExperiment/man/assay-functions.html" class="external-link">compactAssay</a></span><span class="op">(</span><span class="va">ingenes</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RaggedExperiment/man/assay-functions.html" class="external-link">sparseAssay</a></span><span class="op">(</span><span class="va">ingenes</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="mutations-in-genes">Mutations in genes<a class="anchor" aria-label="anchor" href="#mutations-in-genes"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># https://gdac.broadinstitute.org/runs/analyses__2016_01_28/reports/cancer/BRCA-TP/CopyNumberLowPass_Gistic2/nozzle.html</span></span>
<span><span class="va">mre</span> <span class="op">&lt;-</span> <span class="va">BRCA</span><span class="op">[[</span><span class="st">"BRCA_Mutation-20160128"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">## possible bug -- seqlevelsStyle must go first, then genome</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevelsStyle.html" class="external-link">seqlevelsStyle</a></span><span class="op">(</span><span class="va">mre</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"UCSC"</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in (function (seqlevels, genome, new_style) : cannot switch some</span></span>
<span><span class="co">## GRCh37's seqlevels from NCBI to UCSC style</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">genome</a></span><span class="op">(</span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">mre</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"hg19"</span></span>
<span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">mre</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 90490 ranges and 0 metadata columns:</span></span>
<span><span class="co">##           seqnames    ranges strand</span></span>
<span><span class="co">##              &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">##       [1]    chr10 116247760      +</span></span>
<span><span class="co">##       [2]    chr12  43944926      +</span></span>
<span><span class="co">##       [3]     chr3  85932472      +</span></span>
<span><span class="co">##       [4]     chr2  25678299      +</span></span>
<span><span class="co">##       [5]    chr17  40272381      +</span></span>
<span><span class="co">##       ...      ...       ...    ...</span></span>
<span><span class="co">##   [90486]     chr3  48299430      +</span></span>
<span><span class="co">##   [90487]    chr19  52394623      +</span></span>
<span><span class="co">##   [90488]    chr16  30537313      +</span></span>
<span><span class="co">##   [90489]    chr19  35449130      +</span></span>
<span><span class="co">##   [90490]    chr19  53994951      +</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 26 sequences from hg19 genome; no seqlengths</span></span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mingenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">mre</span>, <span class="va">ugenes</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RaggedExperiment/man/assay-functions.html" class="external-link">compactAssay</a></span><span class="op">(</span><span class="va">mingenes</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RaggedExperiment/man/assay-functions.html" class="external-link">sparseAssay</a></span><span class="op">(</span><span class="va">mingenes</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="obtaining-data-directly-from-rtcgatoolbox">Obtaining data directly from <code>RTCGAToolbox</code><a class="anchor" aria-label="anchor" href="#obtaining-data-directly-from-rtcgatoolbox"></a>
</h3>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://mksamur.github.io/RTCGAToolbox/" class="external-link">RTCGAToolbox</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/RTCGAToolbox/man/getLinks.html" class="external-link">getLinks</a></span><span class="op">(</span><span class="st">"BRCA"</span>, CNASeq <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "https://gdac.broadinstitute.org/runs/stddata__2016_01_28/data/BRCA/20160128/gdac.broadinstitute.org_BRCA.Merge_cna__illuminahiseq_dnaseqc__hms_harvard_edu__Level_3__segmentation__seg.Level_3.2016012800.0.0.tar.gz"</span></span></code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">BRCAseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RTCGAToolbox/man/getFirehoseData.html" class="external-link">getFirehoseData</a></span><span class="op">(</span><span class="st">"BRCA"</span>, CNASeq <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## RTCGAToolbox cache directory set to:</span></span>
<span><span class="co">##     /home/mr148/.cache/R/RTCGAToolbox</span></span></code></pre>
<pre><code><span><span class="co">## Using locally cached version of /home/mr148/.cache/R/RTCGAToolbox/20160128-BRCA-Clinical.txt</span></span></code></pre>
<pre><code><span><span class="co">## Using locally cached version of /home/mr148/.cache/R/RTCGAToolbox/20160128-BRCA-CNAseq.txt</span></span></code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cnatoolbox</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RTCGAToolbox/man/biocExtract.html" class="external-link">biocExtract</a></span><span class="op">(</span><span class="va">BRCAseq</span>, <span class="st">"CNASeq"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## working on: CNASeq</span></span></code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevelsStyle.html" class="external-link">seqlevelsStyle</a></span><span class="op">(</span><span class="va">cnatoolbox</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"UCSC"</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">genome</a></span><span class="op">(</span><span class="va">cnatoolbox</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"hg19"</span></span>
<span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">cnatoolbox</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 5298 ranges and 0 metadata columns:</span></span>
<span><span class="co">##          seqnames              ranges strand</span></span>
<span><span class="co">##             &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">##      [1]     chr1       10209-2583075      *</span></span>
<span><span class="co">##      [2]     chr1   2583076-249240606      *</span></span>
<span><span class="co">##      [3]     chr2     10002-243189359      *</span></span>
<span><span class="co">##      [4]     chr3     60175-162511435      *</span></span>
<span><span class="co">##      [5]     chr3 162511436-162626067      *</span></span>
<span><span class="co">##      ...      ...                 ...    ...</span></span>
<span><span class="co">##   [5294]    chr20      60001-62965506      *</span></span>
<span><span class="co">##   [5295]    chr21    9422166-48119869      *</span></span>
<span><span class="co">##   [5296]    chr22   16051206-51244552      *</span></span>
<span><span class="co">##   [5297]    chr23   2699503-116067549      *</span></span>
<span><span class="co">##   [5298]    chr24    2649450-28784074      *</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 24 sequences from hg19 genome; no seqlengths</span></span></code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/RTCGAToolbox/man/getLinks.html" class="external-link">getLinks</a></span><span class="op">(</span><span class="st">"BRCA"</span>, Mutation <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "https://gdac.broadinstitute.org/runs/stddata__2016_01_28/data/BRCA/20160128/gdac.broadinstitute.org_BRCA.Mutation_Packager_Calls.Level_3.2016012800.0.0.tar.gz"</span></span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">BRCAmut</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RTCGAToolbox/man/getFirehoseData.html" class="external-link">getFirehoseData</a></span><span class="op">(</span><span class="st">"BRCA"</span>, Mutation <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## RTCGAToolbox cache directory set to:</span></span>
<span><span class="co">##     /home/mr148/.cache/R/RTCGAToolbox</span></span></code></pre>
<pre><code><span><span class="co">## Using locally cached version of /home/mr148/.cache/R/RTCGAToolbox/20160128-BRCA-Clinical.txt</span></span></code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">muttoolbox</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RTCGAToolbox/man/biocExtract.html" class="external-link">biocExtract</a></span><span class="op">(</span><span class="va">BRCAmut</span>, <span class="st">"Mutation"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## working on: Mutation</span></span></code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevelsStyle.html" class="external-link">seqlevelsStyle</a></span><span class="op">(</span><span class="va">muttoolbox</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"UCSC"</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in (function (seqlevels, genome, new_style) : cannot switch some</span></span>
<span><span class="co">## GRCh37's seqlevels from NCBI to UCSC style</span></span></code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">genome</a></span><span class="op">(</span><span class="va">muttoolbox</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"hg19"</span></span>
<span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">muttoolbox</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 90490 ranges and 0 metadata columns:</span></span>
<span><span class="co">##           seqnames    ranges strand</span></span>
<span><span class="co">##              &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">##       [1]    chr10 116247760      +</span></span>
<span><span class="co">##       [2]    chr12  43944926      +</span></span>
<span><span class="co">##       [3]     chr3  85932472      +</span></span>
<span><span class="co">##       [4]     chr2  25678299      +</span></span>
<span><span class="co">##       [5]    chr17  40272381      +</span></span>
<span><span class="co">##       ...      ...       ...    ...</span></span>
<span><span class="co">##   [90486]     chr3  48299430      +</span></span>
<span><span class="co">##   [90487]    chr19  52394623      +</span></span>
<span><span class="co">##   [90488]    chr16  30537313      +</span></span>
<span><span class="co">##   [90489]    chr19  35449130      +</span></span>
<span><span class="co">##   [90490]    chr19  53994951      +</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 26 sequences from hg19 genome; no seqlengths</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="sizes-for-cnaseq-from-rtcgatoolbox">Sizes for CNAseq from <code>RTCGAToolbox</code><a class="anchor" aria-label="anchor" href="#sizes-for-cnaseq-from-rtcgatoolbox"></a>
</h3>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="va">cnatoolbox</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="va">BRCAseq</span><span class="op">@</span><span class="va">CNASeq</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RaggedExperiment/man/assay-functions.html" class="external-link">sparseAssay</a></span><span class="op">(</span><span class="va">cnatoolbox</span>, sparse <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RaggedExperiment/man/assay-functions.html" class="external-link">compactAssay</a></span><span class="op">(</span><span class="va">cnatoolbox</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RaggedExperiment/man/assay-functions.html" class="external-link">sparseAssay</a></span><span class="op">(</span><span class="va">cnatoolbox</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="va">muttoolbox</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="va">BRCAmut</span><span class="op">@</span><span class="va">Mutation</span><span class="op">)</span></span>
<span><span class="co"># object_size(sparseAssay(muttoolbox, sparse = TRUE)) # typeof character</span></span>
<span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RaggedExperiment/man/assay-functions.html" class="external-link">compactAssay</a></span><span class="op">(</span><span class="va">muttoolbox</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RaggedExperiment/man/assay-functions.html" class="external-link">sparseAssay</a></span><span class="op">(</span><span class="va">muttoolbox</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="restrict-to-genic-regions">Restrict to genic regions<a class="anchor" aria-label="anchor" href="#restrict-to-genic-regions"></a>
</h3>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">incnabox</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">cnatoolbox</span>, <span class="va">ugenes</span><span class="op">)</span></span>
<span><span class="va">inmutbox</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">muttoolbox</span>, <span class="va">ugenes</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="va">incnabox</span><span class="op">)</span></span>
<span><span class="co"># NA</span></span>
<span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RaggedExperiment/man/assay-functions.html" class="external-link">sparseAssay</a></span><span class="op">(</span><span class="va">incnabox</span>, sparse <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RaggedExperiment/man/assay-functions.html" class="external-link">compactAssay</a></span><span class="op">(</span><span class="va">incnabox</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RaggedExperiment/man/assay-functions.html" class="external-link">sparseAssay</a></span><span class="op">(</span><span class="va">incnabox</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="va">inmutbox</span><span class="op">)</span></span>
<span><span class="co"># NA</span></span>
<span><span class="co"># object_size(sparseAssay(inmutbox, sparse = TRUE))</span></span>
<span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RaggedExperiment/man/assay-functions.html" class="external-link">compactAssay</a></span><span class="op">(</span><span class="va">inmutbox</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/object_size.html">object_size</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RaggedExperiment/man/assay-functions.html" class="external-link">sparseAssay</a></span><span class="op">(</span><span class="va">inmutbox</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="table-of-object-sizes-by-data-type-and-source">Table of object sizes by data type and source<a class="anchor" aria-label="anchor" href="#table-of-object-sizes-by-data-type-and-source"></a>
</h2>
<table class="table">
<colgroup>
<col width="11%">
<col width="6%">
<col width="9%">
<col width="14%">
<col width="10%">
<col width="13%">
<col width="20%">
<col width="14%">
</colgroup>
<thead><tr class="header">
<th>Data Source</th>
<th>Assay</th>
<th>Data Type</th>
<th>RaggedExperiment</th>
<th>as.data.frame</th>
<th>sparse Matrix</th>
<th>matrix (reduced rows)</th>
<th>matrix (sparse)</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>curatedTCGAData</td>
<td>CNASeq</td>
<td>numeric</td>
<td>0.2 MB</td>
<td>0.3 MB</td>
<td>0.3 MB</td>
<td>1 MB</td>
<td>1.9 MB</td>
</tr>
<tr class="even">
<td>curatedTCGAData</td>
<td>CNASeq (in genes)</td>
<td>numeric</td>
<td>0.2 MB</td>
<td>0.2 MB</td>
<td>0.3 MB</td>
<td>0.9 MB</td>
<td>1.7 MB</td>
</tr>
<tr class="odd">
<td>curatedTCGAData</td>
<td>Mutation</td>
<td>character</td>
<td>70.6 MB</td>
<td>71.8 MB</td>
<td>NA</td>
<td>680.3 MB</td>
<td>726.2 MB</td>
</tr>
<tr class="even">
<td>curatedTCGAData</td>
<td>Mutation (in genes)</td>
<td>character</td>
<td>37.6 MB</td>
<td>38.1 MB</td>
<td>NA</td>
<td>351.3 MB</td>
<td>375.5 MB</td>
</tr>
</tbody>
</table>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Marcel Ramos.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
